src/js/components/Select/SelectContainer.js

```diff
@@ -39,9 +39,27 @@ const SelectOption = styled(Button)`
   width: 100%;
 `;
 
+const ClearButton = ({ clear, onClear, theme }) => {
+  const { label, position, render } = clear;
+  const align = position !== 'bottom' ? 'start' : 'center';
+  if (render) {
+    return render({ onClear });
+  }
+  return (
+    <Box flex={false}>
+      <Button tabIndex="-1" onClick={onClear}>
+        <Box {...theme.select.clear.container} align={align}>
+          <Text {...theme.select.clear.text}>{label || 'Clear selection'}</Text>
```

Comment:
How would this work with I18N strings? Is it generic enough? 
I think we should be consistent using the `messages` prop for I18N supported strings.

Reply:
Only clear selection wouldn't be able to be translated. But anything they put into label will work. 

The reason this is named render is because it works like the render prop for other grommet components. I believe this operates much differently than form's messages prop.

Reply:
We should use the `messages` prop to every string we are providing. 'Clear selection' is one of them.

Reply:
label would handle that if they need to translate the string. Clear selection is just a default label, if no label is passed in. Same as required is the default label for the error message. I could rename the label prop to messages, but I think that would make the prop unclear as to its use. Since, it operates the same same Button's label prop.

Reply:
Can you take a look at how it's done on other components? I'm pretty sure it should be under `messages`.

Reply:
https://v2.grommet.io/button#label
https://v2.grommet.io/menu#label
https://v2.grommet.io/textarea#placeholder

label makes the most sense, since it's purpose is to work the similarly to Button's label prop. If you're concerned about i18n, any string that I can pass in can be translated it doesn't have to be in the messages prop. The reason messages was necessary when using form was because there was no other way to modify the default required and invalid error responses otherwise. 

here are examples of i18n working with different grommet props.
https://github.com/grommet/grommet-i18n/tree/master/src/js/pages/Reacti18next
The select example has buttons with their labels being translated
https://github.com/grommet/grommet-i18n/blob/master/src/js/pages/Reacti18next/SelectI18next/index.js

Reply:
I think `label` makes sense here. I can see cases where the clear label text should align with the context of the Select. For example, if the Select is in a FormField with `label="Cities"`, the clear option label could be `Clear cities`. In other words, the label could be aligned with the field label, and different for each field.

Reply:
the button label is now either label, 'clear selection', or 'clear {name of formfield}' if its passed in.